File: thrift_python/client/ServiceClientPy.py
import json

from thrift import Thrift
from thrift.protocol import TBinaryProtocol
from thrift.transport import TSocket, TTransport

from server.genPy import DIYFrameworkService
from server.genPy.ttypes import Request, ServiceException


def main():
    # Make socket
    transport = TSocket.TSocket('127.0.0.1', 9090)

    # Buffering is critical. Raw sockets are very slow
    transport = TTransport.TBufferedTransport(transport)
    #transport = TTransport.TFramedTransport(transport)

    # Wrap in a protocol
    protocol = TBinaryProtocol.TBinaryProtocol(transport)

    # Create a client to use the protocol encoder
    client = DIYFrameworkService.Client(protocol)
    transport.open()

    request = Request(None, "predict")
    paramDict = {
        'param': -10
    }

    paramJson = json.dumps(paramDict)
    paramBytes = bytes(paramJson, 'utf-8')
    request.paramJSON = paramBytes

    try:
        response = client.send(request)
        print(response.responseJSON)
    except ServiceException as e:
        print(e.exceptionMess)


if __name__ == '__main__':
    try:
        main()
    except Thrift.TException as tx:
        print('%s' % tx.message)

File: thrift_java/src/main/java/com/fdu/client/ServiceClientPy.java
package com.fdu.client;

import com.alibaba.fastjson.JSON;
import com.alibaba.fastjson.JSONObject;
import com.fdu.server.genJava.generalize.DIYFrameworkService;
import com.fdu.server.genJava.generalize.Request;
import com.fdu.server.genJava.generalize.Response;
import com.fdu.server.genJava.generalize.ServiceException;
import org.apache.thrift.TException;
import org.apache.thrift.protocol.TBinaryProtocol;
import org.apache.thrift.protocol.TProtocol;
import org.apache.thrift.transport.TSocket;
import org.apache.thrift.transport.TTransport;
import org.apache.thrift.transport.TTransportException;

import java.util.HashMap;
import java.util.Map;

public class ServiceClientPy {
    /**
     * 启动客户端远程调用Python服务
     */
    public static void main(String[] args) {
        System.out.println("客户端启动....");
        TTransport transport = null;
        try {
            transport = new TSocket("127.0.0.1", 9090);  // Python Server
            transport.open();

            // 协议要和服务端一致
            TProtocol protocol = new  TBinaryProtocol(transport);

            DIYFrameworkService.Client client = new DIYFrameworkService.Client(protocol);

            // 创建请求对象Request
            Request request = new Request(null, "predict"); // Python request

            Map<String,Integer> map = new HashMap<String,Integer>();
            map.put("param", -10);
            String jsonString = JSON.toJSONString(map);
            byte[] bytes = jsonString.getBytes();
            request.setParamJSON(bytes);

            Response response = client.send(request);
            System.out.println(response.getResponeCode().getValue());
            String responseString = new String(response.getResponseJSON());
            System.out.println(JSONObject.parseObject(responseString));

        } catch (ServiceException e) {
            System.out.println(e.getExceptionMess());
        } catch (TTransportException e) {
            e.printStackTrace();
        } catch (TException te) {
            te.printStackTrace();
        } finally {
            if (null != transport) {
                transport.close();
            }
        }
    }
}